test/cctest/test_sockaddr.cc

```diff
@@ -55,3 +56,34 @@ TEST(SocketAddress, SocketAddressIPv6) {
   addr.set_flow_label(12345);
   CHECK_EQ(addr.flow_label(), 12345);
 }
+
+TEST(SocketAddressBlockList, SocketAddressBlockList) {
+  SocketAddressBlockList block_list;
+
+  sockaddr_storage storage[6];
+
+  SocketAddress::ToSockAddr(AF_INET, "123.123.123.123", 0, &storage[0]);
+  SocketAddress::ToSockAddr(AF_INET, "1.1.1.1", 0, &storage[1]);
+  SocketAddress::ToSockAddr(AF_INET, "1.1.1.10", 0, &storage[2]);
+  SocketAddress::ToSockAddr(AF_INET, "168.172.2.1", 0, &storage[3]);
+  SocketAddress::ToSockAddr(AF_INET6, "::ffff:7b7b:7b7b", 0, &storage[4]);
+  SocketAddress::ToSockAddr(AF_INET6, "::1", 0, &storage[5]);
+
+  SocketAddress addr1(reinterpret_cast<const sockaddr*>(&storage[0]));
+  SocketAddress addr2(reinterpret_cast<const sockaddr*>(&storage[1]));
+  SocketAddress addr3(reinterpret_cast<const sockaddr*>(&storage[2]));
+  SocketAddress addr4(reinterpret_cast<const sockaddr*>(&storage[3]));
+  SocketAddress addr5(reinterpret_cast<const sockaddr*>(&storage[4]));
+  SocketAddress addr6(reinterpret_cast<const sockaddr*>(&storage[5]));
+
+//  block_list.AddSocketAddress(addr1);
+  block_list.AddSocketAddress(addr5);
+//  block_list.AddSocketAddress(addr6);
+//  block_list.AddSocketAddressRange(addr2, addr3);
+//  block_list.AddSocketAddressMask(addr4, 24);
+
+  printf(">>> %d\n", block_list.size());
+
+  printf(">>> %d\n", block_list.Apply(addr1));  // true!
+  printf(">>> %d\n", block_list.Apply(addr5));  // true!
```

Comment:
This test is incomplete and still in development, so for now please ignore the printfs